### PR TITLE
Unit Details Approval

### LINKED ISSUE #24: Simple Process Tracking: Landlord Form Submission Review Flow
# Overview

## Frontend

Implement only the pending/approve banners and the Approve button on the Unit Specific Page - Housing Locator view ([Figma](https://www.figma.com/file/06vEDN5gtlHlC5hxNXMb4g/USHS-Designs?node-id=286%3A17098&mode=dev)).

### Unit API

In  `frontend/src/api/unit.ts` create a `approveUnit` function that takes in a Unit id as a parameter that calls `PUT /api/unit/:id/approve` appropriately.

### Functionality 

If the user is an HL, and the unit is not approved, show the Pending Approval banner at the top and the Approve Listing button at the bottom.

When the Approve Listing button at the bottom is clicked:
1. Call `approveUnit` appropriately.
2. Hide the Approve Listing button
3. Show the Approval Confirmed banner at the top

### Unit Approval Route

Implement the `PUT /api/unit/:id/approve` route ([reference](https://docs.google.com/document/d/19EBAxzbCyPdT6RgELrGsq6A2RPeDt2OSi5DYbFWm03k/edit#bookmark=id.vxe5bt27ec6q))
 
## Acceptance Criteria
- [ ] Unit Specific Page - Housing Locator view implementation
- [ ] `PUT /api/unit/:id/approve` route
- [ ] `approveUnit` API route function
- [ ] Working landlord form submission review flow

### PR DESCRIPTION
**Pull request by YashRavipati1**
## Tracking Info

Resolves #24 
## Changes

Added unit approval functionality on backend and added approval features on unit details page

## Testing

Connected approval route to approve button on unit details page. When clicking approve, the approved parameter on MongoDB for the unit becomes True and the frontend banner changes.

## Confirmation of Change

Screenshot 2024-03-07 at 10 51 42 AM
Screenshot 2024-03-07 at 11 40 40 AM

### CODE REVIEW

File: backend/src/routes/units.ts
Comment (petabite) [inline backend/src/routes/units.ts]: Can we gate this route using the `requireHousingLocator` middleware?

File: backend/src/validators/units.ts
Comment (petabite) [inline backend/src/validators/units.ts]: Is this used anywhere?

**Review by petabite — CHANGES_REQUESTED** [review]
Nicee 🔥 💯 . Just a small thing with auth middleware and you're good to merge after fixing the merge conflicts.

**Review by PoliteUnicorn — COMMENTED** [review]
I think it looks good! I also made an edit button, so there may be some merge conflicts later on, but it looks great!

File: backend/src/routes/units.ts
Comment (YashRavipati1) [reply backend/src/routes/units.ts]: Updated

File: backend/src/validators/units.ts
Comment (YashRavipati1) [reply backend/src/validators/units.ts]: Removed this, I made it to use but then realized we didn't need it because of the formatting of the request but forgot to remove it

**Review by petabite — APPROVED** [review]
Good to merge after the merge conflicts are fixed! 🚢

**Review by PoliteUnicorn — APPROVED** [review]
Looks great! Sorry for the late approval.

**Pull request merged**
